File: Arrays/k_th_smallest.py
n = int(input("Enter the number of elements of the array = "))
array = []
k = int(input("Enter the value of k = "))
print("Elements of the array: ")
for i in range(0,n):
    ele = int(input())
    array.append(ele)
print("Array before sorting: ")
print(array)
for i in range(0,n):
    for j in range(0,n-i-1):
        if(array[j]>array[j+1]):
            t = array[j]
            array[j]=array[j+1]
            array[j+1]=t
print("Array after sorting: ")
print(array)
print("The ", k , " th smallest element in given array is ", array[k-1])
print("The ",k ," th largest element in given array is ",array[n-k])

File: c++/Arrays/k_th_smallest.cpp
#include<iostream>
using namespace std;

int main()
{
	int n,k;
	cout<<"Size of the array = ";
	cin >> n;
	int arr[n];
	cout<<"Value of k = ";
	cin >> k;
	cout<<"Enter the elements of the array \n";
	for(int i=0;i<n;i++)
		cin>>arr[i];
	for(int i=0;i<n;i++)
	{
		for(int j=0;j<n-i-1;j++)
		{
			if(arr[j] > arr[j+1])
			{
				int temp = arr[j];
				arr[j] = arr[j+1];
				arr[j+1] = temp;
			}
		}
	}
	cout<<"\nThe "<<k<<"th smallest number is = "<<arr[k-1];
	cout<<"\nThe "<<k<<"th largest number is  = "<<arr[n-k];
}
